cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -441,34 +480,20 @@ func (ks *kube2sky) updateService(oldObj, newObj interface{}) {
 	ks.newService(newObj)
 }
 
-func newEtcdClient(etcdServer string) (*etcd.Client, error) {
-	var (
-		client *etcd.Client
-		err    error
-	)
-	for attempt := 1; attempt <= maxConnectAttempts; attempt++ {
-		if _, err = etcdutil.GetEtcdVersion(etcdServer); err == nil {
-			break
-		}
-		if attempt == maxConnectAttempts {
-			break
-		}
-		glog.Infof("[Attempt: %d] Attempting access to etcd after 5 second sleep", attempt)
-		time.Sleep(5 * time.Second)
-	}
-	if err != nil {
-		return nil, fmt.Errorf("failed to connect to etcd server: %v, error: %v", etcdServer, err)
-	}
-	glog.Infof("Etcd server found: %v", etcdServer)
-
+func newEtcdClient(config *etcdstorage.EtcdConfig) (client etcd.Client, err error) {
+	ctx := context.TODO()
 	// loop until we have > 0 machines && machines[0] != ""
-	poll, timeout := 1*time.Second, 10*time.Second
+	poll, timeout := pollInterval, pollTimeout
 	if err := wait.Poll(poll, timeout, func() (bool, error) {
-		if client = etcd.NewClient([]string{etcdServer}); client == nil {
-			return false, fmt.Errorf("etcd.NewClient returned nil")
+		if client, err = config.NewEtcdClient(); err != nil {
+			glog.Infof("cannot create etcd client: %v", err)
+			return false, nil
+		}
+		if err = client.Sync(ctx); err != nil {
```

Comment:
I agree that the explicit connectivity verification (as it is here) is probably unnecessary.   
On a side note, I'm a bit concerned about the mutation timeout - if etcd is inaccessible within mutation timeframe (10s by default), kube2sky breaks. Is this a reasonable requirement? Can it not just keep retrying forever?